dotnet/src/SemanticKernel.Core/Functions/KernelFunctionFromPrompt.cs

```diff
@@ -296,11 +305,30 @@ private async Task<(IAIService, PromptExecutionSettings?, string, PromptRendered
 
             if (this._logger.IsEnabled(LogLevel.Trace))
             {
-                this._logger.LogTrace("Rendered prompt changed by handler: {Prompt}", renderedEventArgs.RenderedPrompt);
+                this._logger.LogTrace("Rendered prompt changed by event handler: {Prompt}", renderedEventArgs.RenderedPrompt);
+            }
+        }
+
+        var renderedContext = kernel.OnPromptRenderedFilter(this, arguments, renderedPrompt);
+
+        if (renderedContext is not null &&
+            renderedContext.Cancel is false &&
```

Comment:
```suggestion
           !renderedContext.Cancel &&
```
?

Reply:
Yeah, I put it in similar way as for event args, I think maybe previously this property was nullable, not sure. I simplified in both places now :)